kedro/io/core.py

```diff
@@ -106,19 +106,30 @@ class AbstractDataSet(abc.ABC):
         >>> import pandas as pd
         >>>
         >>> class MyOwnDataSet(AbstractDataSet):
-        >>>     def __init__(self, param1, param2):
+        >>>     def __init__(self, filepath, param1, param2=True):
+        >>>         self._filepath = PurePath(filepath)
         >>>         self._param1 = param1
         >>>         self._param2 = param2
         >>>
         >>>     def _load(self) -> pd.DataFrame:
-        >>>         print("Dummy load: {}".format(self._param1))
-        >>>         return pd.DataFrame()
+        >>>         return pd.read_csv(self._filepath)
         >>>
         >>>     def _save(self, df: pd.DataFrame) -> None:
-        >>>         print("Dummy save: {}".format(self._param2))
+        >>>         df.to_csv(str(self._filepath))
+        >>>
+        >>>     def _exists(self) -> bool:
+        >>>         return path._filepath.is_file()
         >>>
         >>>     def _describe(self):
-        >>>         return dict(param1=self._param1, param2=self._param2)
+        >>>         return dict(    Example catalog.yml specification
+
+    Example catalog.yml specification
+    ::
+        my_dataset:
+            type: <path-to-my-own-dataset>.MyOwnDataSet
+            filepath: data/01_raw/my_data.csv
+            param1: <param1-value> # param1 is a required argument
+            # param2 will be True by defaultparam1=self._param1, param2=self._param2)
```

Comment:
```suggestion
            # param2 will be True by default
```

Reply:
Good Catch, not sure what happened there.